### PR TITLE
Add support for memo field in add funds modal

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/6049

Related Front-end PR: https://github.com/opencollective/opencollective-frontend/pull/8301

### CODE REVIEW

File: server/graphql/v2/object/Order.js [hunk 1]
```diff
@@ -221,6 +221,20 @@ export const Order = new GraphQLObjectType({
           }
         },
       },
+      memo: {
+        type: GraphQLString,
+        description:
+          'Memo field which adds additional details about the order. For example in added funds this can be a note to mark what method (cheque, money order) the funds were received.',
+        async resolve(order, _, { remoteUser }) {
+          const collective = await models.Collective.findOne({ where: { id: order.CollectiveId } });
```
Comment (znarf) [inline server/graphql/v2/object/Order.js:229]: I would even do something like:

```
order.collective = order.collective || (await loaders.Collective.byId.load(order.CollectiveId));
```

Comment (znarf) [reply server/graphql/v2/object/Order.js:229]: Why fetch if it's already available in the order object? And still be defensive so we don't have to think if it's available or not.

[the remark this answers is not in the document]
Comment (SudharakaP) [reply server/graphql/v2/object/Order.js]: Thanks, makes more sense. 👍🏽

Comment (SudharakaP) [reply server/graphql/v2/object/Order.js]: Thanks. This makes more sense. I've refactored this part. 👍🏽

**Review by hdiniz — APPROVED** [review]
lgtm

**Review by Betree — APPROVED** [review]

**Pull request merged**